### PR TITLE
[libc] implement stdc_leading_ones (C23)

### PR DESCRIPTION
**Pull request by nickdesaulniers**
- [libc] implement stdc_leading_ones (C23)
- [libc][test][stdbit] fix dependencies

### CODE REVIEW

File: libc/src/stdbit/stdc_leading_ones_ui.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#ifndef LLVM_LIBC_SRC_STDBIT_STDC_LEADING_ONES_UI_H
+#define LLVM_LIBC_SRC_STDBIT_STDC_LEADING_ONES_UI_H
+
+namespace LIBC_NAMESPACE {
+
+unsigned stdc_leading_ones_ui(unsigned value);
+
+} // namespace LIBC_NAMESPACE
+
+#endif // LLVM_LIBC_SRC_STDBIT_STDC_LEADING_ONES_Ui_H
```
Comment (nickdesaulniers) [inline libc/src/stdbit/stdc_leading_ones_ui.h:18]: lowercase i

Comment (nickdesaulniers) [reply libc/src/stdbit/stdc_leading_ones_ui.h]: fixed in 68052db55a84

File: libc/src/stdbit/stdc_leading_ones_us.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+//===-- Implementation header for stdc_leading_ones_uc ----------*- C++ -*-===//
```
Comment (lntue) [inline libc/src/stdbit/stdc_leading_ones_us.h:1]: nit: `_us` instead of `_uc`.

Comment (nickdesaulniers) [reply libc/src/stdbit/stdc_leading_ones_us.h]: done in 64912cb60bbb

File: libc/src/stdbit/stdc_leading_ones_ui.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+#include "src/stdbit/stdc_leading_ones_ui.h"
+
+#include "src/__support/CPP/bit.h"
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(unsigned, stdc_leading_ones_ui, (unsigned value)) {
+  return cpp::countl_one(value);
```
[the remark this answers is not in the document]
Comment (nickdesaulniers) [reply libc/src/stdbit/stdc_leading_ones_ui.cpp]: done in 58b63c0ec8f4

**Review by lntue — APPROVED** [review]

**Comment by nickdesaulniers** [comment]
Fix: https://github.com/llvm/llvm-project/pull/80318

**Pull request merged**